lms/services/user.py

```diff
@@ -0,0 +1,67 @@
+from lms.models import ApplicationInstance
+from lms.models import LTIUser as User
+from lms.models import display_name as create_display_name
+
+
+class UserService:
+    def __init__(self, db, application_instance_service):
+        self._db = db
+        self._application_instance_service = application_instance_service
+
+    def get(self, oauth_consumer_key, user_id):
+        return (
+            self._db.query(User)
+            .join(ApplicationInstance)
+            .filter(
+                ApplicationInstance.oauth_consumer_key == oauth_consumer_key,
+                User.user_id == user_id,
+            )
+            .one_or_none()
+        )
+
+    def upsert_from_lti(
+        self,
+        oauth_consumer_key,
+        user_id,
+        tool_consumer_instance_guid,
+        roles=None,
+        email=None,
+        display_name=None,
+        lis_person_name_given=None,
+        lis_person_name_family=None,
+        lis_person_name_full=None,
+    ):
+
+        user = self.get(oauth_consumer_key, user_id)
+
+        if not display_name:
+            display_name = create_display_name(
+                lis_person_name_given,
+                lis_person_name_family,
+                lis_person_name_full,
+            )
```

Comment:
I think this is better here instead of spread around the code base. Let the service own the data it manages.